docs/topics/jvm-api-guidelines-predictability.md

```diff
@@ -169,6 +169,9 @@ fun main() { Test.values()[0] = Test.B }
 If you cache values inside the enum, the cache will be corrupted after running the code above. If the values are not 
 cached, it's an additional runtime overhead for each call of the `values()` function.
 
+This is the reason why Kotlin deprecated the `values()` functions since version 1.9 and introduced the `entries()`
```

Comment:
I realized that if the reader wants to learn more about this new `entries()` function and the reasoning behind it, there's no obvious place they can go to as I don't think Kotlin language features are described in blogposts or anything like that. At least I don't see it in the 1.9.0 [announcement post](https://blog.jetbrains.com/kotlin/2023/07/kotlin-1-9-0-released/), even though it was enabled by default in that version.

Very much optional (and it could be seen as excessive information), but it might make sense to include a link to some place that has more information about it, like the [YouTrack ticket](https://youtrack.jetbrains.com/issue/KT-48872/Provide-modern-and-performant-replacement-for-Enum.values) or the [KEEP](https://github.com/Kotlin/KEEP/blob/master/proposals/enum-entries.md).
```suggestion
This is the reason why Kotlin deprecated the `values()` functions since version 1.9 and [introduced](https://youtrack.jetbrains.com/issue/KT-48872/Provide-modern-and-performant-replacement-for-Enum.values) the `entries()`
```

Reply:
Nice idea!